code-challenges/sort/challenges-03.test.js

```diff
@@ -0,0 +1,291 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1
+
+Write a function named sortBackwards that takes in an array of numbers and returns the same array, with the numbers sorted, highest to smallest.
+------------------------------------------------------------------------------------------------ */
+
+const sortBackwards = (arr) => {
+  arr.sort((a, b) => b-a);
+  return arr;
+}; 
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named alphabetize that takes in an array of strings and returns the same array with the strings sorted alphabetically.
+
+In this alphabetization, capital letters come before lower case letters.
+
+For example, ['Alphabet', 'Zebra', 'alphabet', 'carrot'] is correctly sorted.
+------------------------------------------------------------------------------------------------ */
+
+const alphabetize = (arr) => {
+  arr.sort();
+  return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named sortByLength that takes in an array of strings and returns the same array, with the strings sorted by their length, lowest to highest.
+------------------------------------------------------------------------------------------------ */
+
+const sortByLength = (arr) => {
+  arr.sort((a, b) => a.length - b.length);
+  return arr;
+};
+
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named alphabetizeBetter that takes in an array of strings and returns the same array, with the strings sorted alphabetically. Capitalization should not change the sort order of two strings.
+
+For example, ['Alphabet', 'alphabet', 'carrot', 'Zebra'] is correctly sorted, and so is ['alphabet', 'Alphabet', 'carrot', 'Zebra'].
+------------------------------------------------------------------------------------------------ */
+
+const alphabetizeBetter = (arr) => {
+  arr.sort((a, b) => a.toUpperCase() > b.toUpperCase());
+  return arr;
+};
+
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named sortByPrice that takes in an array of objects, each of which has a 'price' property, and sorts those objects by price, lowest to highest, returning the same array.
+
+Here is an example of the input:
+[
+  {name: 'Sweatshirt', price: 45},
+  {name: 'Bookmark', price: 2.50},
+  {name: 'Tote bag', price: 15}
+];
+------------------------------------------------------------------------------------------------ */
+
+const sortByPrice = (arr) => {
+  // arr.sort((a, b) => a.price > b.price);
+  arr.sort((a, b) => {
+    return a.price > b.price;
+  };
```

Comment:
When I try to run tests this javascript file errors out because the above section has not closed out all the code blocks and method calls.